.github/workflows/template_test.yml

```diff
@@ -3,52 +3,48 @@ on:
   workflow_call:
 
 jobs:
-  branch_lint:
-    name: Linting tests
-    runs-on: ubuntu-22.04
-    container:
-      image: nycplanning/build-geosupport:latest
-    defaults:
-      run:
-        shell: bash
-        working-directory: products/template
-    steps:
-      - uses: actions/checkout@v3
-
-      - name: Setup build environment
-        working-directory: ./
-        run: ./bash/docker_container_setup.sh
-
-      - name: Black linting
-        run: black --diff --check .
-
-  branch_unit_tests:
+  unit_tests:
     name: Unit tests
     runs-on: ubuntu-22.04
     env:
       TOY_SECRET: ${{ secrets.TOY_SECRET }}
+      BUILD_ENGINE: postgresql://postgres:postgres@postgis:5432/postgres
+      AWS_S3_ENDPOINT: ${{ secrets.AWS_S3_ENDPOINT }}
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
     container:
       image: nycplanning/build-geosupport:latest
     defaults:
       run:
         shell: bash
         working-directory: products/template
+    services:
+      postgis:
+        image: postgis/postgis:15-3.3-alpine
+        env:
+          POSTGRES_PASSWORD: postgres
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
     steps:
       - uses: actions/checkout@v3
 
-      - name: Setup build environment
-        working-directory: ./
-        run: ./bash/docker_container_setup.sh
-
       - name: Run python tests
         run: |
-          python3 -m pytest --verbose --verbose --cov-config=pyproject.toml --cov .
+          python3 -m pytest -m "not end_to_end" -s --verbose --verbose --durations=0 --cov-config=pyproject.toml --cov .
 
-  branch_build_tests:
-    name: Build test run
+  build:
     uses: ./.github/workflows/template_build.yml
     with:
       run_export: false
       logging_level: DEBUG
       build_note: a test build
     secrets: inherit
+
+  # * end-to-end test must run in build workflow until persistent DB is used
+  # end_to_end_tests:
+  #   name: Run end-to-end tests
```

Comment:
planning to move the test here when the build DB is in edm-data rather than a temporary github action service